File: src/net/eduard/api/lib/game/ItemRandom.java
package net.eduard.api.lib.game;

import net.eduard.api.lib.modules.Extra;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

import net.eduard.api.lib.modules.Mine;

public class ItemRandom {

    private int minAmount = 1;

    private int maxAmount = 1;

    private double chance = 1.0;

    private ItemStack item;

    public ItemRandom() {
    }

    public ItemRandom(ItemStack item, int min, int max) {
        this(item, min, max, 1);
    }

    public ItemRandom(ItemStack item, int min) {
        this(item, min, min);
    }

    public ItemRandom(ItemStack item, int min, int max, double chance) {
        setMinAmount(min);
        setMaxAmount(max);
        setChance(chance);
        setItem(item);
    }

    public ItemStack create() {
        if (Mine.getChance(chance)) {

            ItemStack clone = item.clone();
            int amount = Extra.getRandomInt(getMinAmount(), getMaxAmount());

            clone.setAmount(amount);
            return clone;
        }
        return new ItemStack(Material.AIR);

    }


    public double getChance() {

        return chance;
    }

    public int getMaxAmount() {

        return maxAmount;
    }

    public int getMinAmount() {

        return minAmount;
    }

    public ItemStack getItem() {

        return this.item;
    }

    public void setItem(ItemStack item) {

        this.item = item;
    }

    public void setChance(double chance) {

        this.chance = chance;
    }

    public void setMaxAmount(int maxAmount) {

        this.maxAmount = maxAmount;
    }

    public void setMinAmount(int minAmount) {

        this.minAmount = minAmount;

    }


}

File: src/main/java/net/eduard/api/lib/game/ItemRandom.kt
package net.eduard.api.lib.game

import net.eduard.api.lib.modules.Extra
import net.eduard.api.lib.modules.Mine
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

open class ItemRandom(
    var item: ItemStack? = null,
    var chance: Double=1.0,
    var minAmount: Int=1,
    var maxAmount: Int=1,
    var maxRepeats: Int=1
) {

    constructor(item: ItemStack, chance: Double, minAmount: Int, maxAmount: Int)
            : this(item, chance, minAmount, maxAmount, 1)
    constructor(item: ItemStack, chance: Double, amount: Int)
            : this(item, chance, amount, amount, 1)
    constructor(item: ItemStack, chance: Double)
            : this(item, chance, 1, 1, 1)

    @Transient
    var repeats = 0
    fun create(): ItemStack {
        if (item == null) return ItemStack(Material.AIR)
        if (Extra.getChance(chance)) {
            val clone = item!!.clone()
            clone.amount = Extra.getRandomInt(minAmount, maxAmount)
            repeats++
            return clone
        }
        return ItemStack(Material.AIR)
    }

    fun createRandom(): ItemStack {
        if (item == null) return ItemStack(Material.AIR)
        val clone = item!!.clone()
        clone.amount = Extra.getRandomInt(minAmount, maxAmount)
        repeats++
        return clone
    }
}